test/test_hpyerr.py

```diff
@@ -662,3 +662,23 @@ def do_not_raise_exception(*args):
         with pytest.raises(DummyException):
             mod.f(raise_exception, (DummyException, ), exc_types)
 
+    def test_HPyErr_WriteUnraisable(self, python_subprocess):
+        mod = self.compile_module("""
+            HPyDef_METH(f, "f", f_impl, HPyFunc_NOARGS)
+            static HPy f_impl(HPyContext *ctx, HPy self)
+            {
+                HPyErr_SetString(ctx, ctx->h_ValueError, "error message");
+                HPyErr_WriteUnraisable(ctx, HPy_NULL);
+                return HPyBool_FromLong(ctx, HPyErr_Occurred(ctx));
+            }
+            @EXPORT(f)
+            @INIT
+        """)
+        if not SUPPORTS_SYS_EXECUTABLE:
+            # if sys.executable is not available (e.g. inside pypy app-level)
+            # tests, then skip the rest of this test
+            return
+        # subprocess is not importable in pypy app-level tests
+        result = python_subprocess.run(mod, "mod.f()")
+        assert result.returncode == 0
+
```

Comment:
is it really needed to start a subprocess for this test? You could set `sys.unraisablehook` to something which records it has been invoked, then call `mod.f()` and check that the hook has been actually raised, unless I miss something

Reply:
I didn't know about that.. nice suggestion. Thank you
I noticed that there is support for such routine in `test.support import catch_unraisable_exception` but this has only been added on the `3.8.12` release.. I added it manually until `hpy` drop support for older python releases

Reply:
ah ok, I didn't think about old pythons. +1 for subprocess then

Reply:
approved for merge?